### ISSUE TITLE
I can't get my game connect to google services 

### ISSUE THREAD

**Issue by mobeenk** [playgameservices/play-games-plugin-for-unity#1951]
Hi
I've been suffering with this plugin to get it work for days with no results
first I downloaded the most recent one when I found out later on there is no solution to get it work since it will keep crashing the app,
so I downloaded the 9.33 plugin instead when I build and ran it and try to authenticate I get 
the screen Connecting to Google services then loading icon and afterwards it disappears ... without logging in to anything.
this my first time doing this I appreciate the help please ....

**Comment by drkylec** [comment]
Similar issue. When I try to log in it shows the "connecting to..." then it says "google play games" then I get a load circle for a second and then nothing happens. No errors show up in the console editor. I am using the latest version 0.9.41. This is all on the phone.

**Comment by mobeenk** [comment]
I made a new Test project to test the 9.33 version and it worked by the code of making progress has failed .
it seems that I didn't make the app key correct I'm not sure yet things are still not clear enough to say everything is ok. as for the version 9.41 it keeps crashing the app

**Comment by drkylec** [comment]
I spent a few more days looking into things. I had spent probably in total about a week figure everything out. I had to edit the AndroidManifest in Unity first. 

Then I had to use command lines to get to my KeyStore SHA-1 code from there then put it into my Credentials under the [console.developers.google.com](console.developers.google.com). This was the step that I was missing and was not clear to me from other forums. I had to watch a video on YouTube to figure this one out. Then it was working for me no more crashing or anything. 

I also set my target device a little higher up that someone suggested. So it is now at marshmallow 23. Then my leader board and achievements where not showing up and found that the AndroidManifest was also missing a call to those. So if you have an issue with that this is the code I had to add to it.
`<activity android:name="com.google.games.bridge.NativeBridgeActivity"
           android:theme="@android:style/Theme.Translucent.NoTitleBar.Fullscreen" />`

**Comment by mobeenk** [comment]
Thank you very much for trying to help I Shall try again make a new test project using version 9.41 with the tips you mentioned I hope it works, in case it worked I'll inform you .

**Comment by mobeenk** [comment]
I tried what you said the version 9.41 crashing my game
can you show how you exactly modified the manifest file and the location of it please?

**Comment by drkylec** [comment]
Yeah sure.
`
<?xml version="1.0" encoding="utf-8"?>

<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="(Your Package Name Here)"
    android:versionCode="1"
    android:versionName="1.0" >

    <uses-sdk android:minSdkVersion="23" android:targetSdkVersion="25" />

    <application>

     
        <meta-data android:name="com.google.android.gms.nearby.connection.SERVICE_ID"
            android:value="" />

        
        <meta-data android:name="com.google.android.gms.games.APP_ID"
            android:value="\ (Your Id Here)" />

        
        <meta-data android:name="com.google.android.gms.games.unityVersion"
            android:value="\ 0.9.41" />

        
        <meta-data android:name="com.google.android.gms.version"
            android:value="@integer/google_play_services_version" />
      
      
      <activity android:name="com.google.games.bridge.NativeBridgeActivity"
           android:theme="@android:style/Theme.Translucent.NoTitleBar.Fullscreen" />
    </application>
</manifest>

`
That is the full manifest I have. I made small changes I removed my package name and APP_ID since those are my numbers. The manifest is located under "GooglePlayGames" > "Plugins" > "Android" > "GooglePlayGamesManifest.plugin" > "AndroidManifest.xml"

Also make sure to get your SHA-1 key from your keystore.  If not follow the last part of this guide. [SHA-1 Fingerprint](http://android-er.blogspot.in/2012/12/displaying-sha1-certificate-fingerprint.html)

**Comment by mobeenk** [comment]
It didn't work it kept crashing even that I changed the manifest file like you said 
What version of unity are you using ? mine is 5.4
Would you like to do me a favor and test my sample project and check for me why it's crashing?

**Comment by mobeenk** [comment]
this is the code

    void Start()
    {


        PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder().Build();

        PlayGamesPlatform.InitializeInstance(config);
        // recommended for debugging:
        PlayGamesPlatform.DebugLogEnabled = true;
        // Activate the Google Play Games platform
        PlayGamesPlatform.Activate();
        **signin();**

        /// PlayGamesPlatform.Instance.Authenticate(); // SUCCESS!


    }

    public void signin()
    {
       PlayGamesPlatform.Instance.Authenticate
      //  Social.localUser.Authenticate
       ((bool success) => {
            if (success)
            {
                Debug.Log("yes");
            }
            else if (!success)
            {
                Debug.Log("no");
            }


        });
    }
it crashing at signin()

**Comment by drkylec** [comment]
Yeah sure just upload it somewhere and send me a link to it and I will give it a look.

**Comment by mobeenk** [comment]
Thank you here is the link , I have added a read me file have my email and password of the key
https://files.fm/u/t4685ycr

**Comment by drkylec** [comment]
So I got it to work. It does not crash for me. I added that line to the android manifest from above to get the leaderboard working. I separated the the login and authentication to be off of the button since it will never get called from there. Which was probably why it kept crashing for you. I extracted the keystore number so now that is all set for me anyway you will have to redo this all on your own account in google otherwise I will own the app and your login information.

Side note Unity builds a lot of the files if you ever are sending something just send the asset folder and project settings folder. Also you should never share your keystore information with anyone since it is the only way to prove that you won the app if someone else gets it or you lose the login you lose your app.

**Comment by mobeenk** [comment]
Thanks for reply can you contact me via the email please?
because I didn't figure out how to solve it yet
and what version of unity you are using?

**Comment by MattDownie** [comment]
I've noticed that in my gradle build the game seems to crash unless I remove the space here:


    <meta-data android:name="com.google.android.gms.games.APP_ID"
        android:value="\ (Your Id Here)" />

(after android:value="\ )
Does this actually need to be interpreted as an int?

**Comment by George2514** [comment]
It's interesting to hear your perspective on the game's transformation to 3D. Visual elements, especially graphics and character design, play a significant role in enhancing the overall gaming experience.

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
